scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/TypescriptCodeGenerationService.java

```diff
@@ -183,16 +189,20 @@ static String toString(final TypescriptType type) {
       } else if (type instanceof TSDuration) {
         return "(Temporal.Duration | Discrete<Temporal.Duration>)";
       } else if (type instanceof TSArray t) {
-        final var typeStr = toString(t.elementType());
+        final var typeStr = toString(t.elementType(), true);
         return "((" +typeStr+ ")[] | Discrete<("+typeStr+")[]>)";
       } else if (type instanceof TSStruct t) {
 
         var typeStr = "{ ";
         for(final var keyAndType :  t.keysAndTypes()){
-          typeStr += keyAndType.getLeft() + ":"+ toString(keyAndType.getRight())+",";
+          typeStr += keyAndType.getLeft() + ":"+ toString(keyAndType.getRight(), true)+",";
```

Comment:
If `keysAndTypes` is relatively large, it might be more performant to make `typeStr` a `StringBuilder` and append within the loop.